filetests/isa/intel/prologue-epilogue.cton

```diff
@@ -9,24 +9,14 @@ ebb0:
     return
 }
 
-; check: function %foo(i64 fp [%rbp], i64 csr [%rbx], i64 csr [%r12], i64 csr [%r13], i64 csr [%r14], i64 csr [%r15]) -> i64 fp [%rbp], i64 csr [%rbx], i64 csr [%r12], i64 csr [%r13], i64 csr [%r14], i64 csr [%r15] system_v {
```

Comment:
I've adjusted this test to match what's being output now, but I'm not totally convinced that it's correct. @sunfishcode, can you look at this and make sure it's sane? I struggled for a bit to convince it to use a callee-saved register in a synthetic test here, but with your new pre and post opt passes, that's increasingly hard to do. :)

Reply:
The code looks fine now, and I've also run some non-trivial tests on it and it seems to be working as intended so far. I'll take a closer look at the test here, and I can also help out and add some tests that use callee-saved registers.